src/pykeen/nn/emb.py

```diff
@@ -932,8 +937,8 @@ def forward(
         indices: Optional[torch.LongTensor] = None,
     ) -> torch.FloatTensor:  # noqa: D102
         if indices is None:
-            indices = torch.arange(self.max_id)
-        uniq, inverse = indices.unique(return_inverse=True)
+            indices = torch.arange(self.max_id, device=self.device)
+        uniq, inverse = indices.to(device=self.device).unique(return_inverse=True)
```

Comment:
is the to() operation cheaper or more expensive than a conditional? since the device movement is doubled here if he indices get constructed. Maybe better to make an else with `indices = indices.to(device=self.device)` if indices is None?

Reply:
`.to` is a nop if the tensor is already on the correct device, cf. https://pytorch.org/docs/stable/generated/torch.Tensor.to.html?highlight=#torch.Tensor.to

I am not sure about the overhead, but it _should_ be negligible.